Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabSystemScriptingHandler.cpp

```diff
@@ -68,40 +106,44 @@ namespace AzToolsFramework::Prefab
             }
         }
 
-        bool result = false;
-        [[maybe_unused]] AZ::EntityId commonRoot;
-        EntityList topLevelEntities;
-        AzToolsFramework::ToolsApplicationRequestBus::BroadcastResult(result, &AzToolsFramework::ToolsApplicationRequestBus::Events::FindCommonRootInactive,
-            entities, commonRoot, &topLevelEntities);
-
+        AZStd::unique_ptr<AZ::Entity> containerEntity = CreateContainerEntityAndParentEntities(entities, filePath);
 
-        auto containerEntity = AZStd::make_unique<AZ::Entity>();
-
-        containerEntity->CreateComponent<Components::EditorLockComponent>();
-        containerEntity->CreateComponent<Components::EditorVisibilityComponent>();
-        containerEntity->CreateComponent<Prefab::EditorPrefabComponent>();
+        auto prefab = m_prefabSystemComponentInterface->CreatePrefab(
+            entities, {}, AZ::IO::PathView(AZStd::string_view(filePath)), AZStd::move(containerEntity));
 
+        if (!prefab)
         {
-            auto transformComponent = containerEntity->CreateComponent<Components::TransformComponent>();
-            // Because procedural prefabs need to be deterministic we need to set the component ID to something unique and non-random
-            // The prefab that references the proc prefab will store a patch that references the transform component by it's component ID
-            // If this ID is not stable, the proc prefab will lose its position, parenting, etc data next time it is regenerated
-            auto hash = TypeHash64(reinterpret_cast<const uint8_t*>(filePath.data()), filePath.length(), AZ::HashValue64{0});
-
-            transformComponent->SetId(static_cast<AZ::ComponentId>(hash));
-
+            AZ_Error("PrefabSystemComponenent", false, "Failed to create prefab %s", filePath.c_str());
+            return InvalidTemplateId;
         }
 
-        for (AZ::Entity* entity : topLevelEntities)
+        return prefab->GetTemplateId();
+    }
+
+    TemplateId PrefabSystemScriptingHandler::CreatePrefabTemplateWithCustomEntityAliases(
+        const AZStd::unordered_map<AZ::EntityId, AZStd::string>& entityIds, const AZStd::string& filePath)
+    {
+        AZStd::map<AZStd::string, AZ::Entity*> entities;
+        AZStd::vector<AZ::Entity*> entitiesVector;
+        for (const auto& [entityId, entityAlias] : entityIds)
         {
-            AzToolsFramework::Components::TransformComponent* transformComponent =
-                entity->FindComponent<AzToolsFramework::Components::TransformComponent>();
+            AZ::Entity* entity = nullptr;
+            AZ::ComponentApplicationBus::BroadcastResult(entity, &AZ::ComponentApplicationBus::Events::FindEntity, entityId);
 
-            if (transformComponent)
+            AZ_Warning(
+                "PrefabSystemComponent",
```

Comment:
Shouldn't it be "PrefabSystemScriptingHandler"? Same comment for other AZ_Warning and AZ_Error in this file.